assets/src/pages/room/hooks/useMembraneClient.tsx

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useState } from "react";
+import { MembraneWebRTC, Peer, SerializedMediaEvent, TrackContext } from "@membraneframework/membrane-webrtc-js";
+import { Socket } from "phoenix";
+import { TrackMetadata, PeerMetadata, PeersApi } from "./usePeerState";
+import { isTrackEncoding, isTrackType } from "../../types";
+import { SetErrorMessage } from "../RoomPage";
+
+const parseMetadata = (context: TrackContext) => {
+  const type = context.metadata.type;
+  const active = context.metadata.active;
+  return isTrackType(type) ? { type, active } : { active };
+};
+
+type UseSetupResult = {
+  webrtc?: MembraneWebRTC;
+};
+
+// todo extract callbacks
+export const useMembraneClient = (
+  roomId: string,
+  peerMetadata: PeerMetadata,
+  isSimulcastOn: boolean,
+  api: PeersApi,
+  setErrorMessage: SetErrorMessage
+): UseSetupResult => {
+  const [webrtc, setWebrtc] = useState<MembraneWebRTC | undefined>();
+
+  useEffect(() => {
+    const socket = new Socket("/socket");
+    socket.connect();
+    const socketOnCloseRef = socket.onClose(() => cleanUp());
+    const socketOnErrorRef = socket.onError(() => cleanUp());
+
+    const webrtcChannel = socket.channel(`room:${roomId}`, {
+      isSimulcastOn: isSimulcastOn,
+    });
+
+    webrtcChannel.onError((reason) => {
+      console.error("WebrtcChannel error occurred");
+      console.error(reason);
+      setErrorMessage("WebrtcChannel error occurred");
+    });
+    webrtcChannel.onClose(() => {
+      // TODO fix later
+    });
+
+    const webrtc = new MembraneWebRTC({
+      callbacks: {
+        onSendMediaEvent: (mediaEvent: SerializedMediaEvent) => {
+          webrtcChannel.push("mediaEvent", { data: mediaEvent });
+        },
+        onConnectionError: (message) => {
+          console.error("onConnectionError occurred");
+          console.error(message);
+          setErrorMessage(message);
+        },
+        // todo [Peer] -> Peer[] ???
```

Comment:
Is it done?

Reply:
not yet, I have to test it with it. I extract this to another Jira Task

https://membraneframework.atlassian.net/browse/MV-231